src/PKSim.CLI.Core/Services/QualificationRunner.cs

```diff
@@ -239,25 +243,67 @@ private Task performBuildingBlockSwap(Project projectSnapshot, BuildingBlockSwap
       {
-         var (type, name, snapshotPath) = buildingBlockSwap;
-         var referenceSnasphot = await _snapshotTask.LoadSnapshotFromFile<Project>(snapshotPath);
-         if (referenceSnasphot == null)
-            throw new QualificationRunException(CannotLoadSnapshotFromFile(snapshotPath));
+         var (buildingBlockType, name, snapshotPath) = buildingBlockSwap;
+         var referenceSnapshot = await snapshotProjectFromFile(snapshotPath);
+         var typeDisplay = buildingBlockType.ToString();
 
-         var buildiingBlockToUse = referenceSnasphot.BuildingBlockByTypeAndName(type, name);
-         if (buildiingBlockToUse == null)
-            throw new QualificationRunException(CannotFindBuildingBlockInSnapshot(type.ToString(), name, snapshotPath));
+         var buildingBlockToUse = referenceSnapshot.BuildingBlockByTypeAndName(buildingBlockType, name);
+         if (buildingBlockToUse == null)
+            throw new QualificationRunException(CannotFindBuildingBlockInSnapshot(typeDisplay, name, referenceSnapshot.Name));
 
-         var buildingBlock = projectSnapshot.BuildingBlockByTypeAndName(type, name);
+         var buildingBlock = projectSnapshot.BuildingBlockByTypeAndName(buildingBlockType, name);
          if (buildingBlock == null)
-            throw new QualificationRunException(CannotFindBuildingBlockInSnapshot(type.ToString(), name, projectSnapshot.Name));
+            throw new QualificationRunException(CannotFindBuildingBlockInSnapshot(typeDisplay, name, projectSnapshot.Name));
+
+         projectSnapshot.Swap(buildingBlockToUse);
+      }
+
+      private async Task swapSimulationParametersIn(Project projectSnapshot, SimulationParameterSwap simulationParameter)
+      {
+         var (parameterPath, simulationName, snapshotPath) = simulationParameter;
+         var referenceSnapshot = await snapshotProjectFromFile(snapshotPath);
+
+         var referenceSimulation = simulationFrom(referenceSnapshot, simulationName);
+
+         var referenceParameter = referenceSimulation.ParameterByPath(parameterPath);
+         if (referenceParameter == null)
+            throw new QualificationRunException(CannotFindSimulationParameterInSnapshot(parameterPath, simulationName, referenceSnapshot.Name));
 
-         projectSnapshot.Swap(buildiingBlockToUse);
+         simulationParameter.TargetSimulations?.Each(targetSimulationName =>
+         {
+            var targetSimulation = simulationFrom(projectSnapshot,targetSimulationName);
+            targetSimulation.AddOrUpdate(referenceParameter);
+         });
+      }
+
+      private async Task<Project> snapshotProjectFromFile(string snapshotPath)
+      {
+         if (!_snapshotProjectCache.Contains(snapshotPath))
+         {
+            var snapshot = await _snapshotTask.LoadSnapshotFromFile<Project>(snapshotPath);
+            _snapshotProjectCache[snapshotPath] = snapshot ?? throw new QualificationRunException(CannotLoadSnapshotFromFile(snapshotPath));
```

Comment:
Caching projects as well as we may load them from snapshot quite often depending on how many parameter we are swapping